rs/recovery/src/lib.rs

```diff
@@ -885,6 +889,23 @@ impl Recovery {
     }
 }
 
+#[derive(Debug, Clone, Copy, PartialEq, Serialize, Deserialize)]
+pub enum UploadMethod {
+    Local,
+    Remote(IpAddr),
+}
+
+pub fn upload_method_from_str(s: &str) -> RecoveryResult<UploadMethod> {
+    if s == "local" {
+        return Ok(UploadMethod::Local);
+    }
+    Ok(UploadMethod::Remote(IpAddr::V6(
+        Ipv6Addr::from_str(s).map_err(|e| {
+            RecoveryError::UnexpectedError(format!("Unable to parse ipv6 address {:?}", e))
+        })?,
+    )))
+}
+
```

Comment:
I would probably move this into `util.rs`

Reply:
done